RealestateApi/Service/ReferenceDataService.cs

```diff
@@ -45,6 +47,31 @@ public async Task<AddReferenceDataResponseDto> AddReferenceDataAsync(string refD
                 }
             }
             return response;
-        }     
+        }
+        
+        public async Task<ReferenceDataResponseDto> DeleteReferenceDataAsync(string refDataType, int refDataId)
+        {
+            ReferenceDataResponseDto response = new();
```

Comment:
The first step that should be done here, is to verify that the record that we want to delete, exists :) If it doesn't exist, return an error (console.write line for now) 
If yes, you can proceed with the delete.

If the RefData exists, you can send the refData object to your repository, and after deleting it, you can return that object.
This way we just do 1 call to get the ReferenceData instead of 2 :)